packages/@react-stately/data/docs/useAsyncList.mdx

```diff
@@ -0,0 +1,214 @@
+`load` if no `sort` function is given. Passing a separate `sort` function could be useful when implementing client side
+sorting. Using the `sortDescriptor` in `load` is useful when you need to implement server side sorting, which might be
+an API parameter.
+
+### Client side sorting
+
+This example shows how to implement client side sorting by passing a `sort` function to `useAsyncList` and sorting the
+items array.
+
+```tsx
+let list = useAsyncList({
+  async load({signal}) {
+    // Same load function as before
+  },
+  sort({items, sortDescriptor}) {
+    return {
+      items: items.sort((a, b) => {
+        // Compare the items by the sorted column
+        let cmp = a[sortDescriptor.column] < b[sortDescriptor.column] ? -1 : 1;
+
+        // Flip the direction if descending order is specified.
+        if (sortDescriptor.direction === 'descending') {
+          cmp *= -1;
+        }
+
+        return cmp;
+      })
+    };
+  }
+});
+```
+
+### Server side sorting
+
+Server side sorting could be implemented by using the `sortDescriptor` in the `load` function, and passing a 
+parameter to the API.
+
+```tsx
+let list = useAsyncList({
+  async load({signal, sortDescriptor}) {
+    let url = new URL('http://example.com/api');
+    if (sortDescriptor) {
+      url.searchParams.append('sort_key', sortDescriptor.column);
+      url.searchParams.append('sort_direction', sortDescriptor.direction);
+    }
+
+    let res = await fetch(url, {signal});
+    let json = await res.json();
+    return {
+      items: json.results
+    };
+  }
+});
+```
+
+## Pre-selecting items
+
+`useAsyncList` manages selection state for the list in addition to its data. If you need to automatically select items
+during the initial load, you can do so using the `initialSelectedKeys` option or by returning `selectedKeys` from the
+`load` function in addition to `items`.
+
+### Selecting before loading
+
+If you know what keys to select before items are loaded from the server, use the `initialSelectedKeys` option.
+
+```tsx
+let list = useAsyncList({
+  initialSelectedKeys: ['foo', 'bar'],
+  async load({signal}) {
+    // Same load function as before
+  }
+});
+```
+
+### Selecting based on loaded data
+
+If you need to compute which keys to select based on the loaded data, return `selectedKeys` from the `load` function
+in addition to the `items`.
+
+```tsx
+let list = useAsyncList({
+  async load({signal}) {
+    let res = await fetch('http://example.com/api', {signal});
+    let json = await res.json();
+
+    // Return items and compute selectedKeys based on the data and return a list of ids.
+    return {
+      items: json.results,
+      selectedKeys: json.results.filter(item => item.isSelected).map(item => item.id)
+    };
+  }
+});
+```
+
+## Client side updates
+
+All client side updating methods supported by [useListData](useListData.html) are also supported by `useAsyncList`.
```

Comment:
Maybe move the link to "See the docs for [useListData](useListData) for more details." line instead of the first mention.